Lib/test/test_inspect/test_inspect.py

```diff
@@ -1931,57 +2009,224 @@ class Test:
+        class DataDescriptorSub5(DataDescriptorWithSetNone):
+            """with __set__=None (inherited) and __get__"""
+            def __get__(self, *_): pass
+        class DataDescriptorSub6(DataDescriptorWithDeleteOnly):
+            """with __delete__ (inherited) and __get__"""
+            def __get__(self, *_): pass
+        class DataDescriptorSub7(DataDescriptorWithDeleteNone):
+            """with __delete__=None (inherited) and __get__"""
+            def __get__(self, *_): pass
+        class DataDescriptorSub8(DataDescriptorWithSetDelete):
+            """with __set__ and __delete__ (both inherited) and __get__"""
+            def __get__(self, *_): pass
+        class DataDescriptorSub9(MethodDescriptor):
+            """with __get__ (inherited) and __set__"""
+            def __set__(self, *_): pass
+        class DataDescriptorSub10(MethodDescriptor):
+            """with __get__ (inherited) and __set__=None"""
+            __set__ = None
+        class DataDescriptorSub12(MethodDescriptor):
+            """with __get__ (inherited) and __delete__"""
+            def __delete__(self, *_): pass
+        class DataDescriptorSub11(MethodDescriptor):
+            """with __get__ (inherited) and __delete__=None"""
+            __delete__ = None
+        class DataDescriptorSub13(DataDescriptorWithGetSet):
+            """with __get__ and __set__ (both inherited)"""
+        class DataDescriptorSub14(MethodDescriptor,
+                                  DataDescriptorWithDeleteOnly):
+            """with __get__ and __delete__ (both inherited)"""
+        class DataDescriptorSub15(DataDescriptorWithDeleteOnly,
+                                  DataDescriptorWithGetSet):
+            """with __get__, __set__ and __delete__ (all inherited)"""
+
+        for cls in [
+            DataDescriptorWithSetOnly,
+            DataDescriptorWithSetNone,
+            DataDescriptorWithDeleteOnly,
+            DataDescriptorWithDeleteNone,
+            DataDescriptorWithSetDelete,
+            DataDescriptorWithGetSet,
+            DataDescriptorWithGetDelete,
+            DataDescriptorWithGetSetDelete,
+            DataDescriptorSub1,
+            DataDescriptorSub2,
+            DataDescriptorSub3,
+            DataDescriptorSub4,
+            DataDescriptorSub5,
+            DataDescriptorSub6,
+            DataDescriptorSub7,
+            DataDescriptorSub8,
+            DataDescriptorSub9,
+            DataDescriptorSub10,
+            DataDescriptorSub11,
+            DataDescriptorSub12,
+            DataDescriptorSub13,
+            DataDescriptorSub14,
+            DataDescriptorSub15,
+        ]:
+            with self.subTest(f"a descriptor {cls.__doc__}"):
+                self.assertTrue(inspect.isdatadescriptor(cls()))
+            with self.subTest(f"a descriptor class {cls.__doc__} "
+                              f"(not a descriptor itself)"):
+                self.assertFalse(inspect.isdatadescriptor(cls))
```

Comment:
I think it's better to also `cls=cls` to know which class was responsible for the error, not just what you expect:

```py
with self.subTest(f"a descriptor {cls.__doc__}", cls=cls):
```